helix-term/src/commands/typed.rs

```diff
@@ -596,19 +598,37 @@ fn force_cquit(
 fn theme(
     cx: &mut compositor::Context,
     args: &[Cow<str>],
-    _event: PromptEvent,
+    event: PromptEvent,
 ) -> anyhow::Result<()> {
-    let theme = args.first().context("Theme not provided")?;
-    let theme = cx
-        .editor
-        .theme_loader
-        .load(theme)
-        .with_context(|| format!("Failed setting theme {}", theme))?;
-    let true_color = cx.editor.config().true_color || crate::true_color();
-    if !(true_color || theme.is_16_color()) {
-        bail!("Unsupported theme: theme requires true color support");
-    }
-    cx.editor.set_theme(theme);
+    let true_color = cx.editor.config.load().true_color || crate::true_color();
+    match event {
+        PromptEvent::Abort => {
+            cx.editor.unset_theme_preview();
+        }
```

Comment:
Not sure about this, but is this something to be delegated to `Drop` and make the API better? I think this is fine, but not sure if doing ownership would be good here? Not sure if it's even feasible, just thinking.

Reply:
Thought about this for a bit and this is what I've arrived at:

I think the drop would have to be on a TypableCommand, but if that's the case, we would have to create wrapper structs for each command as each one would drop differently. Since drop doesn't have `compositor::Context` passed in, I suppose it would need to be made a field of that struct. At that point I'm not too sure what the lifetimes would look like, but I reckon it's still doable. I think if this were to be done, we would probably want to also place the other event handlers into traits too for consistency.

Reply:
I disagree, it's a stateless function. No reason to involve Drop.